### PR TITLE
Start CMake docs ; Publish Doxygen Internally

### PR DESCRIPTION
**Pull request by ChristianTackeGSI**
1. On Each PR and branch build, run doxygen and publish the result on an internal webserver.
2. Document `fairroot_check_root_cxxstd_compatibility` in the CMake docs.

* The new CMake docs can be seen here: http://lxci001.gsi.de/ci/tmp/jenkins-FairRootGroup-FairRoot-PR-1216-3/d9/dcc/cmake.html#autotoc_md15

---

Checklist:

* [X] Rebased against `dev` branch
* [X] My name is in the resp. CONTRIBUTORS/AUTHORS file
* [X] Followed [the seven rules of great commit messages](https://chris.beams.io/posts/git-commit/#seven-rules)

### CODE REVIEW

**Comment by ChristianTackeGSI** [comment]
@TobiasStockmanns you asked about better CMake docs. Could you take a look whether this is the right direction?

File: FairRoot_doxygen_test.cmake [hunk 1]
```diff
@@ -9,26 +9,17 @@
...
+fairroot_ctest_setup()
 
 if ("$ENV{LABEL}" STREQUAL "")
   set(CTEST_BUILD_NAME "format-check")
 else()
   set(CTEST_BUILD_NAME $ENV{LABEL})
 endif()
 
+set(WEB_PATH "/shared/alfaci/web")
+set(WEB_URL_BASE "http://lxci001.gsi.de/ci")
```
Comment (dennisklein) [inline FairRoot_doxygen_test.cmake:21]: Could we model those as input parameters and set them from the callsite (possibly only set a default value here)?

File: FairRoot_doxygen_test.cmake [hunk 2]
```diff
@@ -42,6 +33,32 @@ list(APPEND options
...
+ctest_build(TARGET DoxygenDoc FLAGS ""
+            NUMBER_ERRORS _ctest_build_num_errs
+            RETURN_VALUE _ctest_build_ret_val)
+
+if(_ctest_build_ret_val OR _ctest_build_num_errs)
+  message(STATUS "Skipping Publishing, because build failed"
+          " (return value: ${_ctest_build_ret_val},"
+          " number of errors: ${_ctest_build_num_errs})")
+else()
+  if (DEFINED ENV{CHANGE_ID} AND NOT ("$ENV{BUILD_TAG}" STREQUAL ""))
```
Comment (dennisklein) [inline FairRoot_doxygen_test.cmake:45]: Now that I see this, I am thinking whether we should better not hardwire jenkins variables here, but define an input parameter, e.g. `IS_PR` and set it from the jenkins script?

Comment (dennisklein) [reply FairRoot_doxygen_test.cmake:45]: Alternatively, lets put at least a comment that explains what those environment variables mean and where they come from?

**Review by dennisklein — APPROVED** [review]

**Pull request merged**